### PR TITLE
Oleksii bozhchenko

### PR DESCRIPTION
**Pull request by abozhchenko**
HW1, HW2, HW3, HW4 - completed
HW5 - in progress

### CODE REVIEW

File: OleksiiBozhchenko/HW1/1.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+answer_name = input("What is your name? : ")
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW1/1.py:1]: good job

File: OleksiiBozhchenko/HW1/2.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+a = int(input("Please input number for variable \"a\": "))
+b = int(input("Please input number for variable \"b\": "))
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW1/2.py:3]: good job


print(f"a + b = {a+b}")

print(f"a - b = {a-b}")

print(f"a * b = {a * b}")

print(f"a / b = {a / b}")

print(f"a ** b = {a ** b}")

print(f"a // b = { a // b}")

print(f"a % b = {a % b")

File: OleksiiBozhchenko/HW3/Task1-1.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+48.Unless explicitly silenced.
+49.In the face of ambiguity, refuse the temptation to guess.
+50.There should be one-- and preferably only one --obvious way to do it.
+51.Although that way may not be obvious at first unless you're Dutch.
+52.Now is better than never.
+53.Although never is often better than *right* now.
+54.If the implementation is hard to explain, it's a bad idea.
+55.If the implementation is easy to explain, it may be a good idea.
+56.Namespaces are one honking great idea -- let's do more of those!"""
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW3/Task1-1.py:59]: good job

File: OleksiiBozhchenko/HW3/Task1-2.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import os
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW3/Task1-2.py:2]: good job

File: OleksiiBozhchenko/HW3/Task2.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
+number = (input("Enter 4 digits: "))
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW3/Task2.py:2]: good job

File: OleksiiBozhchenko/HW3/Task3.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# #
+# # Interchange the values ​​of two variables without using the third variable.
+# #
+a = 5
+b = 6
+print(f"Variable a = {a}")
+print(f"Variable b = {b}")
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW3/Task3.py:8]: good job

File: OleksiiBozhchenko/HW4/ClassWork1.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# Write a script, which of the two entered numbers will determine which of them is more and which is less. 
+# Take into account the case of equality of numbers.
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW4/ClassWork1.py:3]: good job

File: OleksiiBozhchenko/HW4/ClassWork2.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# Write a script that will check whether the entered number is even or odd and display the appropriate message.
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW4/ClassWork2.py:2]: good job

File: OleksiiBozhchenko/HW4/Task1.py [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# Task1. Write a script that will calculate the factorial of the entered number without using recursion.
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW4/Task1.py:2]: good job


input_number = int(input("Input natural number: "))

fact = 1

if input_number == 1 or input_number == 0:
    print("Factorial {}! =".format(input_number), fact)
elif input_number < 0:
    print("Factorial non positive number don't exist")
else:
    for item in range(1,input_number + 1):
        fact *= item
    print("Factorial {}! =".format(input_number), fact)

File: OleksiiBozhchenko/HW5/ClassWork1.py [hunk 10]
```diff
@@ -0,0 +1,16 @@
+# Print all even numbers less than 100 
+# (write two variants of the code: one using the while loop, and the other using the for loop).
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW5/ClassWork1.py:3]: good job

File: OleksiiBozhchenko/HW5/ClassWork2.py [hunk 11]
```diff
@@ -0,0 +1,20 @@
+# Print all odd numbers less than 100. 
+# (write two versions of the code: one using the continue operator, and the other without this operator).
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW5/ClassWork2.py:3]: good job

File: OleksiiBozhchenko/HW5/Task1.py [hunk 12]
```diff
@@ -0,0 +1,21 @@
+# Create a list that contains elements of integer type, 
+# then use loop to change the type of these elements to a floating type
+# (Hint: use the built-in float() function)
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW5/Task1.py:4]: good job

File: OleksiiBozhchenko/HW5/Task2.py [hunk 13]
```diff
@@ -0,0 +1,31 @@
+# Print Fibonacci numbers up to the entered number n, using cycles.
+# (Sequence of Fibonacci numbers: 0, 1, 1, 2, 3, 5, 8, 13, etc.)
+
+# #
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW5/Task2.py:4]: good job

File: OleksiiBozhchenko/HW1/2.py (hunk 2 above)
Comment (abozhchenko) [reply OleksiiBozhchenko/HW1/2.py:3]: ok

File: OleksiiBozhchenko/HW4/Task1.py (hunk 9 above)
Comment (abozhchenko) [reply OleksiiBozhchenko/HW4/Task1.py:2]: Thank You!
This kind of output from print() - new for me, I will study it.

File: OleksiiBozhchenko/HW6/ClassWork.py [hunk 14]
```diff
@@ -0,0 +1,33 @@
+# Create a list of integers that are entered from the terminal and determine the maximum 
+# and minimum number among them.
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW6/ClassWork.py:2]: good job

File: OleksiiBozhchenko/HW6/Task1.py [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+        odd.append(i)
+    else:
+        other.append(i)    
+
+print(f"Even numbers are: {even}\nOdd numbers are: {odd}\nNon divisible numbers are: {other}")
+
+# #
+# # Example 2 
+# # (In this example I experimented with List Comprehension)
+# #
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW6/Task1.py:26]: you output:
Even numbers are: [2, 4, 6, 8]
Odd numbers are: [3,      ??? 6 ???  , 9]
Non divisible numbers are: [1, 5, 7]

another implementation:

l1 = [ x for x in range(1, 11) if x % 2 == 0 ]
l2 = [ x for x in range(1, 11) if x % 2 == 1 and x % 3 == 0 ]
l3 = [ x for x in range(1, 11) if not x % 2 == 0 and not x % 3 == 0 ]
print(l1, 'is even multiple of 2')
print(l2, 'is an odd multiple of 3')
print(l3, 'not divisible by 2 and 3')

File: OleksiiBozhchenko/HW6/Task2.py [hunk 16]
```diff
@@ -0,0 +1,11 @@
+# Write a script that checks the login that the user enters.
+# If the login is "First", then greet the user. 
+# If the login is different, send error message.
+# (need to use loop while) 
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW6/Task2.py:5]: good job

File: OleksiiBozhchenko/HW7/ClassWork1.py [hunk 17]
```diff
@@ -0,0 +1,27 @@
+# Write a function that returns the arithmetic mean of any number of numbers
+
+# #
+# # Example 1
+# #
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW7/ClassWork1.py:5]: good job

File: OleksiiBozhchenko/HW7/Task1.py [hunk 18]
```diff
@@ -0,0 +1,18 @@
+# Write a function that returns the largest number of two nmbers
+# (use DocStrings documentation strings in the function)
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW7/Task1.py:3]: good job

File: OleksiiBozhchenko/HW7/Task2.py [hunk 19]
```diff
@@ -0,0 +1,44 @@
+# Write a program that calculate the area of a rectangle, triangle and circle.
+# Write three function to calculate the area. And call them in the main program
+# depending on the user's choise.
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW7/Task2.py:4]: good job

File: OleksiiBozhchenko/HW7/Task3.py [hunk 20]
```diff
@@ -0,0 +1,11 @@
+# Write a function that calculate thr number of characters included in given string.
+# input: "hello"
+# output: {"h":1,"e":1,"l":2,"o":1}
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW7/Task3.py:4]: here is a more optimal implementation:
def count_symbols_in_row(word):
    result = {}
    for item in word:
        if item in result:
            continue
        else:
            result.update({str(item): word.count(item)})
    return result

File: OleksiiBozhchenko/HW7/coderunner.py [hunk 21]
```diff
@@ -0,0 +1,87 @@
+print("")
+print("Task 1")
+# Given a list of numbers and a value n, write a function that returns the probability of choosing a number greater
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW7/coderunner.py:3]: good job

File: OleksiiBozhchenko/HW7/codewars.py [hunk 22]
```diff
@@ -0,0 +1,175 @@
+# Task 1
+# DESCRIPTION:
+# Jenny has written a function that returns a greeting for a user. 
+# However, she's in love with Johnny, and would like to greet him slightly different. 
+# She added a special case to her function, but she made a mistake.
+def greet(name):
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW7/codewars.py:6]: good job

File: OleksiiBozhchenko/HW8/ClassWork/ClassWork.py [hunk 23]
```diff
@@ -0,0 +1,26 @@
+import operations
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW8/ClassWork/ClassWork.py:1]: good job

File: OleksiiBozhchenko/HW8/ClassWork/operations.py [hunk 24]
```diff
@@ -0,0 +1,11 @@
+def sum(num1, num2):
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW8/ClassWork/operations.py:1]: good work

File: OleksiiBozhchenko/HW8/Task2.py [hunk 25]
```diff
@@ -0,0 +1,49 @@
+# Task 2.
+# Write a Python program to check the validity of a password(input from user).
+# Validation:
+# At least 1 letter between [a-z] and 1 letter between [A-Z]
+# At least 1 number between [0-9]
+# At least 1 character from [@#$]
+# Minimum length 6 characters
+# Maximum length 16 characters
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW8/Task2.py:9]: good job

File: OleksiiBozhchenko/HW8/Task3/Task3.py [hunk 26]
```diff
@@ -0,0 +1,40 @@
+# Write a program that calculates the area of a rectangle S=a*b, the area of a
+# triangle S=0.5*h*a, the area of a circle S=pi*r**2. This module must be used in
+# another module in which we ask the user the area of which figure he wants to
+# calculate.
+# (To perform the task, you need to import the "math" module, and from it the
+# "pow()" function and the value of the variable "pi", and module, which contains
+# three functions for finding areas, into the main program. The basic logic of the
+# program is executed in the main module).
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW8/Task3/Task3.py:9]: good job

File: OleksiiBozhchenko/HW8/Task3/operations.py [hunk 27]
```diff
@@ -0,0 +1,13 @@
+import math
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW8/Task3/operations.py:1]: good job

File: OleksiiBozhchenko/HW9/ClassWork.py [hunk 28]
```diff
@@ -0,0 +1,25 @@
+from pyowm import OWM
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW9/ClassWork.py:1]: good job

File: OleksiiBozhchenko/HW9/Task1.py [hunk 29]
```diff
@@ -0,0 +1,31 @@
+# # Task 1. 
+# # Write a game script that randomly generates a number from a range of 1 to 100 
+# # and asks the user to guess that number in 10 tries.
+# # The program reads the numbers entered by the user and prompts the user whether the guessed number 
+# # is greater or less than the number entered by the user. The game must continue until the user has used 10 attempts 
+# # and guessed the number. If the user guessed the number, the program prints a congratulatory message, 
+# # and if 10 attempts have been exhausted and the user did not have time to guess the number, 
+# # then the corresponding message is displayed. (to perform the task, you need to import the random module,
+# # and from it the randint() function)
+
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW9/Task1.py:10]: good job

File: OleksiiBozhchenko/HW9/Task2.py [hunk 30]
```diff
@@ -0,0 +1,57 @@
+import pygame
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW9/Task2.py:1]: good job

File: OleksiiBozhchenko/HW9/Task3.py [hunk 31]
```diff
@@ -0,0 +1,58 @@
+from pyowm import OWM
```
Comment (kolyasalubov) [inline OleksiiBozhchenko/HW9/Task3.py:1]: good job

**Pull request merged**